manifests/agent/service/systemd.pp

```diff
@@ -0,0 +1,67 @@
+class puppet::agent::service::systemd (
+  $enabled = false,
+) {
+
+  exec { 'systemctl-daemon-reload-puppet':
+    refreshonly => true,
+    path        => $::path,
+    command     => 'systemctl daemon-reload',
+  }
+
+  case $enabled {
+    true: {
+      # Use the same times as for cron
+      $times = ip_to_cron($puppet::runinterval)
+
+      $command = $puppet::systemd_cmd ? {
+        undef   => "/usr/bin/env puppet agent --config ${puppet::dir}/puppet.conf --onetime --no-daemonize --detailed-exitcode --no-usecacheonfailure",
+        default => $puppet::systemd_cmd,
+      }
+
+      file { "/etc/systemd/system/${::puppet::systemd_unit_name}.timer":
+        content => template('puppet/agent/systemd.puppet-run.timer.erb'),
+        notify  => [
+          Exec['systemctl-daemon-reload'],
+          Service['puppet-run.timer'],
+        ],
+      }
+
+      file { "/etc/systemd/system/${::puppet::systemd_unit_name}.service":
+        content => template('puppet/agent/systemd.puppet-run.service.erb'),
+        notify  => Exec['systemctl-daemon-reload-puppet'],
+      }
+
+      service { 'puppet-run.timer':
```

Comment:
Ok, ignore all those comments, my ability to read code is clearly failing. No need to use variables for internal puppet labels I guess ;)